neonvm/runner/main.go

```diff
@@ -673,38 +645,14 @@ func main() {
 		qemuCmd = append(qemuCmd, "-incoming", fmt.Sprintf("tcp:0:%d", vmv1.MigrationPort))
 	}
 
-	selfCgroupPath, err := getSelfCgroupPath(logger)
-	if err != nil {
-		logger.Fatal("Failed to get self cgroup path", zap.Error(err))
-	}
-	// Sometimes we'll get just '/' as our cgroup path. If that's the case, we should reset it so
-	// that the cgroup '/neonvm-qemu-...' still works.
-	if selfCgroupPath == "/" {
-		selfCgroupPath = ""
-	}
-	// ... but also we should have some uniqueness just in case, so we're not sharing a root level
-	// cgroup if that *is* what's happening. This *should* only be relevant for local clusters.
-	//
-	// We don't want to just use the VM spec's .status.PodName because during migrations that will
-	// be equal to the source pod, not this one, which may be... somewhat confusing.
-	cgroupPath := fmt.Sprintf("%s/neonvm-qemu-%s", selfCgroupPath, selfPodName)
-
-	logger.Info("Determined QEMU cgroup path", zap.String("path", cgroupPath))
-
-	if err := setCgroupLimit(logger, qemuCPUs.use, cgroupPath); err != nil {
-		logger.Fatal("Failed to set cgroup limit", zap.Error(err))
-	}
```

Comment:
PR in its current state hasn't replicated this — so, right now there's a regression where a 0.25 CPU VM starts at 1 CPU and needs to scale down.

Probably need to add milli CPU as an arg to container-mgr or something, and then do a similar "set on startup". (and actually, that'd also remove the need for nil-able cpu returned from the runner pod as well, so can revert 5acf53663499c8d16eeae2a388d3a36658a2f0ca)

Reply:
fixed in 2af7af5adcb738d736eac1095f098511b8e691f4

Reply:
I don't understand why we need the new arguments here. Can't we just set the initial limits once `container-mgr` is starting?

Reply:
In theory, yeah. In practice, it seemed like it wasn't simple to make sure that we didn't get any weird transitory states — we'd have to update the runner before moving from Pending to Running, or e.g. if .status.cpus is nil, update the runner before fetching the info.

The PR in its present form (reverted nillable runner CPU; container-mgr has the new arg) doesn't need any of the meat of the reconcile logic to be updated (with the notable exception of #749). That's my primary motivation (because the existing logic is _so complex_).